src/routes/assistants/+page.svelte

```diff
@@ -0,0 +1,77 @@
+<script lang="ts">
+	import { goto } from "$app/navigation";
+	import { base } from "$app/paths";
+	import { page } from "$app/stores";
+	import type { PageData } from "./$types";
+	export let data: PageData;
+	let selectedModel = $page.url.searchParams.get("modelId") ?? "";
+
+	const onModelChange = (e: Event) => {
+		const newUrl = new URL($page.url);
+		if ((e.target as HTMLSelectElement).value === "") {
+			newUrl.searchParams.delete("modelId");
+		} else {
+			newUrl.searchParams.set("modelId", (e.target as HTMLSelectElement).value);
+		}
+		goto(newUrl);
+	};
+</script>
+
+<div class="mx-auto w-full max-w-4xl">
+	<h1 class="mt-10 w-full text-center text-2xl font-bold">Popular assistants</h1>
+	<h3 class="mt-2 w-full text-center text-sm">
+		These are the most popular assistants on the platform.
+	</h3>
+
+	<label class="mt-10 text-sm">
+		Filter by model:
+		<select
+			class="mx-auto mt-2 rounded-md border border-gray-300 pl-1 text-center text-sm dark:border-gray-700 dark:bg-gray-700"
+			bind:value={selectedModel}
+			on:change={onModelChange}
+		>
+			<option value="">All</option>
+			{#each data.models as model}
+				<option value={model.name}>{model.name}</option>
```

Comment:
don't we need something like `<option selected={model.name === selectedModel}>` ? 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option#selected

or is it somehow handled automatically?

Reply:
No need the value is binded to the select `bind:value={selectedModel}` and it picks the selected option from there.